modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridCachePartitionsStateValidationTest.java

```diff
@@ -323,4 +330,139 @@ public void waitUntilAllSingleMessagesAreSent() throws IgniteCheckedException {
             allSingleMessagesSent.get();
         }
     }
+
+    /**
+     * Three tests that partitions state validation works correctly and show partition size always (GG-24471):
+     *  1. Only partition counters are inconsistent.
+     *  2. Only partition size are inconsistent.
+     *  3. Partition counters and size are inconsistent.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testValidationIfPartitionCountersAreInconsistentWithSize() throws Exception {
+        testLogger = new ListeningTestLogger();
+
+        LogListener lsnr = LogListener.matches("Partitions update counters are inconsistent for ").build();
+
+        testLogger.registerListener(lsnr);
+
+        try {
+            IgniteEx ignite = startGrid(0);
```

Comment:
Grids don't have to be started one by one. Use startGrids()

Reply:
The idea is to start only the first node with LogListener, and then two nodes without.